In [9]:
%load_ext sql

The sql extension is already loaded. To reload it, use:
  %reload_ext sql


In [10]:
%sql sqlite:///Chinook_Sqlite.sqlite

In [11]:
%%sql
SELECT * FROM Album LIMIT 10;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1, 'For Those About To Rock We Salute You', 1),
 (2, 'Balls to the Wall', 2),
 (3, 'Restless and Wild', 2),
 (4, 'Let There Be Rock', 1),
 (5, 'Big Ones', 3),
 (6, 'Jagged Little Pill', 4),
 (7, 'Facelift', 5),
 (8, 'Warner 25 Anos', 6),
 (9, 'Plays Metallica By Four Cellos', 7),
 (10, 'Audioslave', 8)]

In [12]:
%%sql
SELECT TrackID,Name,AlbumID
FROM Track LIMIT 10

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1, 'For Those About To Rock (We Salute You)', 1),
 (2, 'Balls to the Wall', 2),
 (3, 'Fast As a Shark', 3),
 (4, 'Restless and Wild', 3),
 (5, 'Princess of the Dawn', 3),
 (6, 'Put The Finger On You', 1),
 (7, "Let's Get It Up", 1),
 (8, 'Inject The Venom', 1),
 (9, 'Snowballed', 1),
 (10, 'Evil Walks', 1)]

In [13]:
%%sql
SELECT album.Title as album_title,
track.name as track_name 
FROM Track track
JOIN Album album ON track.AlbumId = album.AlbumId
LIMIT 10;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[('For Those About To Rock We Salute You', 'For Those About To Rock (We Salute You)'),
 ('Balls to the Wall', 'Balls to the Wall'),
 ('Restless and Wild', 'Fast As a Shark'),
 ('Restless and Wild', 'Restless and Wild'),
 ('Restless and Wild', 'Princess of the Dawn'),
 ('For Those About To Rock We Salute You', 'Put The Finger On You'),
 ('For Those About To Rock We Salute You', "Let's Get It Up"),
 ('For Those About To Rock We Salute You', 'Inject The Venom'),
 ('For Those About To Rock We Salute You', 'Snowballed'),
 ('For Those About To Rock We Salute You', 'Evil Walks')]

In [14]:
%%sql
SELECT * FROM Artist LIMIT 10;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1, 'AC/DC'),
 (2, 'Accept'),
 (3, 'Aerosmith'),
 (4, 'Alanis Morissette'),
 (5, 'Alice In Chains'),
 (6, 'Antônio Carlos Jobim'),
 (7, 'Apocalyptica'),
 (8, 'Audioslave'),
 (9, 'BackBeat'),
 (10, 'Billy Cobham')]

In [15]:
%%sql
Select *
From Invoice LIMIT 10;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1, 2, '2009-01-01 00:00:00', 'Theodor-Heuss-Straße 34', 'Stuttgart', None, 'Germany', '70174', 1.98),
 (2, 4, '2009-01-02 00:00:00', 'Ullevålsveien 14', 'Oslo', None, 'Norway', '0171', 3.96),
 (3, 8, '2009-01-03 00:00:00', 'Grétrystraat 63', 'Brussels', None, 'Belgium', '1000', 5.94),
 (4, 14, '2009-01-06 00:00:00', '8210 111 ST NW', 'Edmonton', 'AB', 'Canada', 'T6G 2C7', 8.91),
 (5, 23, '2009-01-11 00:00:00', '69 Salem Street', 'Boston', 'MA', 'USA', '2113', 13.86),
 (6, 37, '2009-01-19 00:00:00', 'Berger Straße 10', 'Frankfurt', None, 'Germany', '60316', 0.99),
 (7, 38, '2009-02-01 00:00:00', 'Barbarossastraße 19', 'Berlin', None, 'Germany', '10779', 1.98),
 (8, 40, '2009-02-01 00:00:00', '8, Rue Hanovre', 'Paris', None, 'France', '75002', 1.98),
 (9, 42, '2009-02-02 00:00:00', '9, Place Louis Barthou', 'Bordeaux', None, 'France', '33000', 3.96),
 (10, 46, '2009-02-03 00:00:00', '3 Chatham Street', 'Dublin', 'Dublin', 'Ireland', None, 5.94)]

In [16]:
%%sql
Select Playlistid,
Name
From Playlist LIMIT 10;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1, 'Music'),
 (2, 'Movies'),
 (3, 'TV Shows'),
 (4, 'Audiobooks'),
 (5, '90’s Music'),
 (6, 'Audiobooks'),
 (7, 'Movies'),
 (8, 'Music'),
 (9, 'Music Videos'),
 (10, 'TV Shows')]

In [17]:
%%sql
Select CustomerId,
InvoiceDate, 
BillingCity 
From Invoice LIMIT 10;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(2, '2009-01-01 00:00:00', 'Stuttgart'),
 (4, '2009-01-02 00:00:00', 'Oslo'),
 (8, '2009-01-03 00:00:00', 'Brussels'),
 (14, '2009-01-06 00:00:00', 'Edmonton'),
 (23, '2009-01-11 00:00:00', 'Boston'),
 (37, '2009-01-19 00:00:00', 'Frankfurt'),
 (38, '2009-02-01 00:00:00', 'Berlin'),
 (40, '2009-02-01 00:00:00', 'Paris'),
 (42, '2009-02-02 00:00:00', 'Bordeaux'),
 (46, '2009-02-03 00:00:00', 'Dublin')]

In [18]:
%%sql
Select TrackId, 
GenreId, 
Composer, 
UnitPrice 
From Track LIMIT 10

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 0.99),
 (2, 1, None, 0.99),
 (3, 1, 'F. Baltes, S. Kaufman, U. Dirkscneider & W. Hoffman', 0.99),
 (4, 1, 'F. Baltes, R.A. Smith-Diesel, S. Kaufman, U. Dirkscneider & W. Hoffman', 0.99),
 (5, 1, 'Deaffy & R.A. Smith-Diesel', 0.99),
 (6, 1, 'Angus Young, Malcolm Young, Brian Johnson', 0.99),
 (7, 1, 'Angus Young, Malcolm Young, Brian Johnson', 0.99),
 (8, 1, 'Angus Young, Malcolm Young, Brian Johnson', 0.99),
 (9, 1, 'Angus Young, Malcolm Young, Brian Johnson', 0.99),
 (10, 1, 'Angus Young, Malcolm Young, Brian Johnson', 0.99)]

In [19]:
%%sql
Select DISTINCT PlaylistId 
From PlaylistTrack 

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1,),
 (3,),
 (5,),
 (8,),
 (9,),
 (10,),
 (11,),
 (12,),
 (13,),
 (14,),
 (15,),
 (16,),
 (17,),
 (18,)]

In [20]:
%%sql
Select PlaylistId,TrackId 
From PlaylistTrack LIMIT 10;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1, 3402),
 (1, 3389),
 (1, 3390),
 (1, 3391),
 (1, 3392),
 (1, 3393),
 (1, 3394),
 (1, 3395),
 (1, 3396),
 (1, 3397)]

In [21]:
%%sql
Select *
From MediaType
Limit 50;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1, 'MPEG audio file'),
 (2, 'Protected AAC audio file'),
 (3, 'Protected MPEG-4 video file'),
 (4, 'Purchased AAC audio file'),
 (5, 'AAC audio file')]

In [22]:
%%sql
Select PlaylistId 
From PlaylistTrack
WHERE PlaylistId>2 LIMIT 10;


 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(3,), (3,), (3,), (3,), (3,), (3,), (3,), (3,), (3,), (3,)]

In [23]:
%%sql
Select DISTINCT PlaylistId 
From PlaylistTrack
WHERE PlaylistId<>2 LIMIT 10;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1,), (3,), (5,), (8,), (9,), (10,), (11,), (12,), (13,), (14,)]

In [24]:
%%sql
Select COUNT(PlaylistId) 
From PlaylistTrack
WHERE PlaylistId;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(8715,)]

In [25]:
%%sql
Select DISTINCT PlaylistId 
From PlaylistTrack
WHERE PlaylistId
BETWEEN 0 AND 4;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1,), (3,)]

In [26]:
%%sql
SELECT * FROM Album WHERE Title IS NULL;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[]

In [27]:
%%sql
SELECT * FROM Album LIMIT 10;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1, 'For Those About To Rock We Salute You', 1),
 (2, 'Balls to the Wall', 2),
 (3, 'Restless and Wild', 2),
 (4, 'Let There Be Rock', 1),
 (5, 'Big Ones', 3),
 (6, 'Jagged Little Pill', 4),
 (7, 'Facelift', 5),
 (8, 'Warner 25 Anos', 6),
 (9, 'Plays Metallica By Four Cellos', 7),
 (10, 'Audioslave', 8)]

In [28]:
%%sql
SELECT * FROM ALbum WHERE ArtistId IN (1,2,3)

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1, 'For Those About To Rock We Salute You', 1),
 (4, 'Let There Be Rock', 1),
 (2, 'Balls to the Wall', 2),
 (3, 'Restless and Wild', 2),
 (5, 'Big Ones', 3)]

In [29]:
%%sql
SELECT * FROM ALbum WHERE ArtistId==1;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1, 'For Those About To Rock We Salute You', 1), (4, 'Let There Be Rock', 1)]

In [30]:
%%sql
SELECT * FROM ALbum WHERE AlbumId IN (1,4)

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1, 'For Those About To Rock We Salute You', 1), (4, 'Let There Be Rock', 1)]

In [31]:
%%sql
SELECT * FROM ALbum WHERE AlbumId = 1 OR AlbumId = 4;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1, 'For Those About To Rock We Salute You', 1), (4, 'Let There Be Rock', 1)]

In [32]:
%%sql
SELECT * FROM Album WHERE(ArtistId=2 OR ArtistId=8) AND AlbumId>2 LIMIT 50;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(3, 'Restless and Wild', 2),
 (10, 'Audioslave', 8),
 (11, 'Out Of Exile', 8),
 (271, 'Revelations', 8)]

In [33]:
%%sql
SELECT * FROM Album WHERE(ArtistId=2 OR ArtistId=8) AND AlbumId<>10 LIMIT 50;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(2, 'Balls to the Wall', 2),
 (3, 'Restless and Wild', 2),
 (11, 'Out Of Exile', 8),
 (271, 'Revelations', 8)]

In [34]:
%%sql
SELECT * FROM Album WHERE Title LIKE '%Wall%';

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(2, 'Balls to the Wall', 2)]

In [35]:
%%sql
SELECT * FROM Album WHERE Title LIKE 'Bal%';

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(2, 'Balls to the Wall', 2)]

In [36]:
%%sql
SELECT * FROM Album WHERE Title LIKE '%Wall';

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(2, 'Balls to the Wall', 2)]

Statement with wildcards will take longer to run if used at the end of the search patterns.Thats why placement of wildcards is important

In [37]:
%%sql
SELECT * FROM Album ORDER BY AlbumId DESC LIMIT 10 ;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(347, 'Koyaanisqatsi (Soundtrack from the Motion Picture)', 275),
 (346, 'Mozart: Chamber Music', 274),
 (345, "Monteverdi: L'Orfeo", 273),
 (344, "Schubert: The Late String Quartets & String Quintet (3 CD's)", 272),
 (343, 'Respighi:Pines of Rome', 226),
 (342, 'Locatelli: Concertos for Violin, Strings and Continuo, Vol. 3', 271),
 (341, 'Great Recordings of the Century - Shubert: Schwanengesang, 4 Lieder', 270),
 (340, "Liszt - 12 Études D'Execution Transcendante", 269),
 (339, "Great Recordings of the Century: Paganini's 24 Caprices", 268),
 (338, 'Nielsen: The Six Symphonies', 267)]

In [38]:
%%sql
SELECT * FROM Album ORDER BY AlbumId ASC LIMIT 10 ;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1, 'For Those About To Rock We Salute You', 1),
 (2, 'Balls to the Wall', 2),
 (3, 'Restless and Wild', 2),
 (4, 'Let There Be Rock', 1),
 (5, 'Big Ones', 3),
 (6, 'Jagged Little Pill', 4),
 (7, 'Facelift', 5),
 (8, 'Warner 25 Anos', 6),
 (9, 'Plays Metallica By Four Cellos', 7),
 (10, 'Audioslave', 8)]

In [39]:
%%sql
.tables

 * sqlite:///Chinook_Sqlite.sqlite
(sqlite3.OperationalError) near ".": syntax error
[SQL: .tables]
(Background on this error at: https://sqlalche.me/e/20/e3q8)


AGGRAGATE FUNCTION
    * AVG() ,
    *COUNT() ,
    *MIN() ,
    *MAX() ,
    *SUM() ,

In [40]:
%%sql
SELECT * FROM InvoiceLine ORDER BY InvoiceLineId DESC LIMIT 50 ;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(2240, 412, 3177, 1.99, 1),
 (2239, 411, 3163, 0.99, 1),
 (2238, 411, 3154, 0.99, 1),
 (2237, 411, 3145, 0.99, 1),
 (2236, 411, 3136, 0.99, 1),
 (2235, 411, 3127, 0.99, 1),
 (2234, 411, 3118, 0.99, 1),
 (2233, 411, 3109, 0.99, 1),
 (2232, 411, 3100, 0.99, 1),
 (2231, 411, 3091, 0.99, 1),
 (2230, 411, 3082, 0.99, 1),
 (2229, 411, 3073, 0.99, 1),
 (2228, 411, 3064, 0.99, 1),
 (2227, 411, 3055, 0.99, 1),
 (2226, 411, 3046, 0.99, 1),
 (2225, 410, 3037, 0.99, 1),
 (2224, 410, 3031, 0.99, 1),
 (2223, 410, 3025, 0.99, 1),
 (2222, 410, 3019, 0.99, 1),
 (2221, 410, 3013, 0.99, 1),
 (2220, 410, 3007, 0.99, 1),
 (2219, 410, 3001, 0.99, 1),
 (2218, 410, 2995, 0.99, 1),
 (2217, 410, 2989, 0.99, 1),
 (2216, 409, 2983, 0.99, 1),
 (2215, 409, 2979, 0.99, 1),
 (2214, 409, 2975, 0.99, 1),
 (2213, 409, 2971, 0.99, 1),
 (2212, 409, 2967, 0.99, 1),
 (2211, 409, 2963, 0.99, 1),
 (2210, 408, 2959, 0.99, 1),
 (2209, 408, 2957, 0.99, 1),
 (2208, 408, 2955, 0.99, 1),
 (2207, 408, 2953, 0.99, 1),
 (2206, 407, 2951, 0.99, 1),
 (2205, 407, 2949, 0.99, 1),
 (2204, 406, 2947, 0.99, 1),
 (2203, 406, 2946, 0.99, 1),
 (2202, 405, 2945, 0.99, 1),
 (2201, 404, 2931, 0.99, 1),
 (2200, 404, 2922, 1.99, 1),
 (2199, 404, 2913, 1.99, 1),
 (2198, 404, 2904, 1.99, 1),
 (2197, 404, 2895, 1.99, 1),
 (2196, 404, 2886, 1.99, 1),
 (2195, 404, 2877, 1.99, 1),
 (2194, 404, 2868, 1.99, 1),
 (2193, 404, 2859, 1.99, 1),
 (2192, 404, 2850, 1.99, 1),
 (2191, 404, 2841, 1.99, 1)]

In [41]:
%%sql
SELECT InvoiceLineId,InvoiceId,TrackId,UnitPrice,Quantity,round((UnitPrice*Quantity)*1.5,2) AS TOTAL_COST_WITH_VAT FROM InvoiceLine ORDER BY InvoiceLineId DESC LIMIT 10 ;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(2240, 412, 3177, 1.99, 1, 2.99),
 (2239, 411, 3163, 0.99, 1, 1.49),
 (2238, 411, 3154, 0.99, 1, 1.49),
 (2237, 411, 3145, 0.99, 1, 1.49),
 (2236, 411, 3136, 0.99, 1, 1.49),
 (2235, 411, 3127, 0.99, 1, 1.49),
 (2234, 411, 3118, 0.99, 1, 1.49),
 (2233, 411, 3109, 0.99, 1, 1.49),
 (2232, 411, 3100, 0.99, 1, 1.49),
 (2231, 411, 3091, 0.99, 1, 1.49)]

In [42]:
%%sql
SELECT AVG(round((UnitPrice*Quantity)*1.5,2)) AS AVG_COST_WITH_VAT FROM InvoiceLine ;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1.5643303571427714,)]

In [43]:
%%sql
SELECT AVG(UnitPrice) AS Average_unit_price FROM InvoiceLine ;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1.0395535714285522,)]

In [44]:
%%sql
SELECT COUNT(*) AS TOTAL_ROW FROM ALbum 

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(347,)]

In [45]:
%%sql
SELECT COUNT(*) AS TOTAL_CUSTOMER FROM Customer;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(59,)]

In [46]:
%%sql
SELECT MIN(UnitPrice) AS minimum_unit_price FROM InvoiceLine ;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(0.99,)]

In [47]:
%%sql
SELECT MAX(UnitPrice) AS maximum_unit_price FROM InvoiceLine ;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1.99,)]

In [48]:
%%sql
SELECT SUM(UnitPrice) AS total_99_unit_price FROM InvoiceLine WHERE UnitPrice=0.99 ;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(2107.710000000004,)]

In [49]:
%%sql
SELECT SUM(UnitPrice) AS total_1_99_unit_price FROM InvoiceLine WHERE UnitPrice=1.99 ;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(220.89000000000027,)]

In [50]:
%%sql
SELECT COUNT(UnitPrice) AS total_99_unit_price FROM InvoiceLine WHERE UnitPrice=0.99 ;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(2129,)]

In [51]:
%%sql
SELECT COUNT(UnitPrice) AS total_99_unit_price FROM InvoiceLine WHERE UnitPrice=1.99 ;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(111,)]

CAN NOT USE DISTINCT() on COUNT(*)

In [52]:
%%sql
SELECT COUNT(*) AS total_customer_listed FROM Invoice ;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(412,)]

In [53]:
%%sql
SELECT * FROM Invoice LIMIT 10;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1, 2, '2009-01-01 00:00:00', 'Theodor-Heuss-Straße 34', 'Stuttgart', None, 'Germany', '70174', 1.98),
 (2, 4, '2009-01-02 00:00:00', 'Ullevålsveien 14', 'Oslo', None, 'Norway', '0171', 3.96),
 (3, 8, '2009-01-03 00:00:00', 'Grétrystraat 63', 'Brussels', None, 'Belgium', '1000', 5.94),
 (4, 14, '2009-01-06 00:00:00', '8210 111 ST NW', 'Edmonton', 'AB', 'Canada', 'T6G 2C7', 8.91),
 (5, 23, '2009-01-11 00:00:00', '69 Salem Street', 'Boston', 'MA', 'USA', '2113', 13.86),
 (6, 37, '2009-01-19 00:00:00', 'Berger Straße 10', 'Frankfurt', None, 'Germany', '60316', 0.99),
 (7, 38, '2009-02-01 00:00:00', 'Barbarossastraße 19', 'Berlin', None, 'Germany', '10779', 1.98),
 (8, 40, '2009-02-01 00:00:00', '8, Rue Hanovre', 'Paris', None, 'France', '75002', 1.98),
 (9, 42, '2009-02-02 00:00:00', '9, Place Louis Barthou', 'Bordeaux', None, 'France', '33000', 3.96),
 (10, 46, '2009-02-03 00:00:00', '3 Chatham Street', 'Dublin', 'Dublin', 'Ireland', None, 5.94)]

groupby column always work with the aggragate function it ans the question
BASED ON WHAT ARE YOU AGGRARATING?

In [54]:
%%sql
SELECT * FROM Invoice LIMIT 10;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1, 2, '2009-01-01 00:00:00', 'Theodor-Heuss-Straße 34', 'Stuttgart', None, 'Germany', '70174', 1.98),
 (2, 4, '2009-01-02 00:00:00', 'Ullevålsveien 14', 'Oslo', None, 'Norway', '0171', 3.96),
 (3, 8, '2009-01-03 00:00:00', 'Grétrystraat 63', 'Brussels', None, 'Belgium', '1000', 5.94),
 (4, 14, '2009-01-06 00:00:00', '8210 111 ST NW', 'Edmonton', 'AB', 'Canada', 'T6G 2C7', 8.91),
 (5, 23, '2009-01-11 00:00:00', '69 Salem Street', 'Boston', 'MA', 'USA', '2113', 13.86),
 (6, 37, '2009-01-19 00:00:00', 'Berger Straße 10', 'Frankfurt', None, 'Germany', '60316', 0.99),
 (7, 38, '2009-02-01 00:00:00', 'Barbarossastraße 19', 'Berlin', None, 'Germany', '10779', 1.98),
 (8, 40, '2009-02-01 00:00:00', '8, Rue Hanovre', 'Paris', None, 'France', '75002', 1.98),
 (9, 42, '2009-02-02 00:00:00', '9, Place Louis Barthou', 'Bordeaux', None, 'France', '33000', 3.96),
 (10, 46, '2009-02-03 00:00:00', '3 Chatham Street', 'Dublin', 'Dublin', 'Ireland', None, 5.94)]

In [55]:
%%sql
SELECT BillingCountry,COUNT(CustomerId) AS TOTAL_CUSTOMER FROM Invoice GROUP BY BillingCountry;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[('Argentina', 7),
 ('Australia', 7),
 ('Austria', 7),
 ('Belgium', 7),
 ('Brazil', 35),
 ('Canada', 56),
 ('Chile', 7),
 ('Czech Republic', 14),
 ('Denmark', 7),
 ('Finland', 7),
 ('France', 35),
 ('Germany', 28),
 ('Hungary', 7),
 ('India', 13),
 ('Ireland', 7),
 ('Italy', 7),
 ('Netherlands', 7),
 ('Norway', 7),
 ('Poland', 7),
 ('Portugal', 14),
 ('Spain', 7),
 ('Sweden', 7),
 ('USA', 91),
 ('United Kingdom', 21)]

In [56]:
%%sql 
SELECT * FROM Invoice LIMIT 30;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1, 2, '2009-01-01 00:00:00', 'Theodor-Heuss-Straße 34', 'Stuttgart', None, 'Germany', '70174', 1.98),
 (2, 4, '2009-01-02 00:00:00', 'Ullevålsveien 14', 'Oslo', None, 'Norway', '0171', 3.96),
 (3, 8, '2009-01-03 00:00:00', 'Grétrystraat 63', 'Brussels', None, 'Belgium', '1000', 5.94),
 (4, 14, '2009-01-06 00:00:00', '8210 111 ST NW', 'Edmonton', 'AB', 'Canada', 'T6G 2C7', 8.91),
 (5, 23, '2009-01-11 00:00:00', '69 Salem Street', 'Boston', 'MA', 'USA', '2113', 13.86),
 (6, 37, '2009-01-19 00:00:00', 'Berger Straße 10', 'Frankfurt', None, 'Germany', '60316', 0.99),
 (7, 38, '2009-02-01 00:00:00', 'Barbarossastraße 19', 'Berlin', None, 'Germany', '10779', 1.98),
 (8, 40, '2009-02-01 00:00:00', '8, Rue Hanovre', 'Paris', None, 'France', '75002', 1.98),
 (9, 42, '2009-02-02 00:00:00', '9, Place Louis Barthou', 'Bordeaux', None, 'France', '33000', 3.96),
 (10, 46, '2009-02-03 00:00:00', '3 Chatham Street', 'Dublin', 'Dublin', 'Ireland', None, 5.94),
 (11, 52, '2009-02-06 00:00:00', '202 Hoxton Street', 'London', None, 'United Kingdom', 'N1 5LH', 8.91),
 (12, 2, '2009-02-11 00:00:00', 'Theodor-Heuss-Straße 34', 'Stuttgart', None, 'Germany', '70174', 13.86),
 (13, 16, '2009-02-19 00:00:00', '1600 Amphitheatre Parkway', 'Mountain View', 'CA', 'USA', '94043-1351', 0.99),
 (14, 17, '2009-03-04 00:00:00', '1 Microsoft Way', 'Redmond', 'WA', 'USA', '98052-8300', 1.98),
 (15, 19, '2009-03-04 00:00:00', '1 Infinite Loop', 'Cupertino', 'CA', 'USA', '95014', 1.98),
 (16, 21, '2009-03-05 00:00:00', '801 W 4th Street', 'Reno', 'NV', 'USA', '89503', 3.96),
 (17, 25, '2009-03-06 00:00:00', '319 N. Frances Street', 'Madison', 'WI', 'USA', '53703', 5.94),
 (18, 31, '2009-03-09 00:00:00', '194A Chain Lake Drive', 'Halifax', 'NS', 'Canada', 'B3S 1C5', 8.91),
 (19, 40, '2009-03-14 00:00:00', '8, Rue Hanovre', 'Paris', None, 'France', '75002', 13.86),
 (20, 54, '2009-03-22 00:00:00', '110 Raeburn Pl', 'Edinburgh ', None, 'United Kingdom', 'EH4 1HH', 0.99),
 (21, 55, '2009-04-04 00:00:00', '421 Bourke Street', 'Sidney', 'NSW', 'Australia', '2010', 1.98),
 (22, 57, '2009-04-04 00:00:00', 'Calle Lira, 198', 'Santiago', None, 'Chile', None, 1.98),
 (23, 59, '2009-04-05 00:00:00', '3,Raj Bhavan Road', 'Bangalore', None, 'India', '560001', 3.96),
 (24, 4, '2009-04-06 00:00:00', 'Ullevålsveien 14', 'Oslo', None, 'Norway', '0171', 5.94),
 (25, 10, '2009-04-09 00:00:00', 'Rua Dr. Falcão Filho, 155', 'São Paulo', 'SP', 'Brazil', '01007-010', 8.91),
 (26, 19, '2009-04-14 00:00:00', '1 Infinite Loop', 'Cupertino', 'CA', 'USA', '95014', 13.86),
 (27, 33, '2009-04-22 00:00:00', '5112 48 Street', 'Yellowknife', 'NT', 'Canada', 'X1A 1N6', 0.99),
 (28, 34, '2009-05-05 00:00:00', 'Rua da Assunção 53', 'Lisbon', None, 'Portugal', None, 1.98),
 (29, 36, '2009-05-05 00:00:00', 'Tauentzienstraße 8', 'Berlin', None, 'Germany', '10789', 1.98),
 (30, 38, '2009-05-06 00:00:00', 'Barbarossastraße 19', 'Berlin', None, 'Germany', '10779', 3.96)]

In [57]:
%%sql
SELECT COUNT(CustomerID) AS NUMBER_OF_CUSTOMER, BillingCountry  FROM Invoice GROUP BY BillingCountry HAVING COUNT(CustomerID) > 30;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(35, 'Brazil'), (56, 'Canada'), (35, 'France'), (91, 'USA')]

In [58]:
%%sql
SELECT COUNT(*) NUMBER_OF_COUNTRY FROM Invoice WHERE BillingCountry="Brazil"

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(35,)]

In [59]:
%%sql
SELECT COUNT(*) NUMBER_OF_COUNTRY FROM Invoice WHERE BillingCountry="Canada"

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(56,)]

In [60]:
%%sql
SELECT COUNT(*) NUMBER_OF_COUNTRY FROM Invoice WHERE BillingCountry="USA"

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(91,)]

In [61]:
%%sql
SELECT COUNT(*) NUMBER_OF_COUNTRY FROM Invoice WHERE BillingCountry="Germany"

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(28,)]

In [62]:
%%sql
SELECT COUNT(CustomerID) AS NUMBER_OF_CUSTOMER, BillingCountry  FROM Invoice GROUP BY BillingCountry HAVING COUNT(CustomerID) > 30 ORDER BY NUMBER_OF_CUSTOMER DESC;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(91, 'USA'), (56, 'Canada'), (35, 'France'), (35, 'Brazil')]

-- SELECT 
-- distinct Extended_step 
SELECT distinct Extended_step
FROM salary_range_by_job_classification;

-- Select 
-- min(Biweekly_high_Rate)
-- From salary_range_by_job_classification
-- WHERE Biweekly_high_Rate <> '$0.00'    
SELECT DISTINCT Biweekly_high_Rate FROM salary_range_by_job_classification
ORDER BY Biweekly_high_Rate ASC;

Select
job_code,
pay_type
FROM
salary_range_by_job_classification WHERE job_code LIKE '03%' 

SELECT Step FROM salary_range_by_job_classification WHERE job_code = '0110' OR job_code = '0400';
SELECT (Biweekly_High_Rate-Biweekly_Low_Rate) FROM salary_range_by_job_classification WHERE Job_Code='0170';
SELECT Extended_Step FROM salary_range_by_job_classification WHERE Pay_Type IN ('M','H','D');
SELECT * FROM salary_range_by_job_classification WHERE Union_Code='990' AND (Salary_SetID = 'COMMN' OR Salary_SetID = 'SFMTA' );
SELECt * FROM Tracks WHERE Milliseconds > 5000000 ;
SELECT COUNT(*) FROM Invoices WHERE Total BETWEEN 5 AND 15;
SELECt * FROM Customers WHERE State IN ('RJ', 'DF', 'AB', 'BC', 'CA', 'WA', 'NY');
SELECT * FROM Invoices WHERE InvoiceId = 315;
SELECT * FROM Customers WHERE Email LIKE 'J%@gmail.com';
SELECT * FROM Invoices WHERE BillingCity IN ('Brasília', 'Edmonton','Vancouver') ORDER BY InvoiceId ;
SELECT CustomerId,COUNT(*) FROM Invoices GROUP BY CustomerId;
SELECT Albums.ALbumId,Albums.title,Tracks.TrackId FROM Albums
join Tracks ON Albums.ALbumId = Tracks.AlbumId
GROUP BY Albums.AlbumId
HAVING COUNT(Tracks.TrackId)>=12

In [76]:
%%sql
SELECT * FROM Album;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1, 'For Those About To Rock We Salute You', 1),
 (2, 'Balls to the Wall', 2),
 (3, 'Restless and Wild', 2),
 (4, 'Let There Be Rock', 1),
 (5, 'Big Ones', 3),
 (6, 'Jagged Little Pill', 4),
 (7, 'Facelift', 5),
 (8, 'Warner 25 Anos', 6),
 (9, 'Plays Metallica By Four Cellos', 7),
 (10, 'Audioslave', 8),
 (11, 'Out Of Exile', 8),
 (12, 'BackBeat Soundtrack', 9),
 (13, 'The Best Of Billy Cobham', 10),
 (14, 'Alcohol Fueled Brewtality Live! [Disc 1]', 11),
 (15, 'Alcohol Fueled Brewtality Live! [Disc 2]', 11),
 (16, 'Black Sabbath', 12),
 (17, 'Black Sabbath Vol. 4 (Remaster)', 12),
 (18, 'Body Count', 13),
 (19, 'Chemical Wedding', 14),
 (20, 'The Best Of Buddy Guy - The Millenium Collection', 15),
 (21, 'Prenda Minha', 16),
 (22, 'Sozinho Remix Ao Vivo', 16),
 (23, 'Minha Historia', 17),
 (24, 'Afrociberdelia', 18),
 (25, 'Da Lama Ao Caos', 18),
 (26, 'Acústico MTV [Live]', 19),
 (27, 'Cidade Negra - Hits', 19),
 (28, 'Na Pista', 20),
 (29, 'Axé Bahia 2001', 21),
 (30, 'BBC Sessions [Disc 1] [Live]', 22),
 (31, 'Bongo Fury', 23),
 (32, 'Carnaval 2001', 21),
 (33, 'Chill: Brazil (Disc 1)', 24),
 (34, 'Chill: Brazil (Disc 2)', 6),
 (35, 'Garage Inc. (Disc 1)', 50),
 (36, 'Greatest Hits II', 51),
 (37, 'Greatest Kiss', 52),
 (38, 'Heart of the Night', 53),
 (39, 'International Superhits', 54),
 (40, 'Into The Light', 55),
 (41, 'Meus Momentos', 56),
 (42, 'Minha História', 57),
 (43, 'MK III The Final Concerts [Disc 1]', 58),
 (44, 'Physical Graffiti [Disc 1]', 22),
 (45, 'Sambas De Enredo 2001', 21),
 (46, 'Supernatural', 59),
 (47, 'The Best of Ed Motta', 37),
 (48, 'The Essential Miles Davis [Disc 1]', 68),
 (49, 'The Essential Miles Davis [Disc 2]', 68),
 (50, 'The Final Concerts (Disc 2)', 58),
 (51, "Up An' Atom", 69),
 (52, 'Vinícius De Moraes - Sem Limite', 70),
 (53, 'Vozes do MPB', 21),
 (54, 'Chronicle, Vol. 1', 76),
 (55, 'Chronicle, Vol. 2', 76),
 (56, 'Cássia Eller - Coleção Sem Limite [Disc 2]', 77),
 (57, 'Cássia Eller - Sem Limite [Disc 1]', 77),
 (58, 'Come Taste The Band', 58),
 (59, 'Deep Purple In Rock', 58),
 (60, 'Fireball', 58),
 (61, "Knocking at Your Back Door: The Best Of Deep Purple in the 80's", 58),
 (62, 'Machine Head', 58),
 (63, 'Purpendicular', 58),
 (64, 'Slaves And Masters', 58),
 (65, 'Stormbringer', 58),
 (66, 'The Battle Rages On', 58),
 (67, "Vault: Def Leppard's Greatest Hits", 78),
 (68, 'Outbreak', 79),
 (69, 'Djavan Ao Vivo - Vol. 02', 80),
 (70, 'Djavan Ao Vivo - Vol. 1', 80),
 (71, 'Elis Regina-Minha História', 41),
 (72, 'The Cream Of Clapton', 81),
 (73, 'Unplugged', 81),
 (74, 'Album Of The Year', 82),
 (75, 'Angel Dust', 82),
 (76, 'King For A Day Fool For A Lifetime', 82),
 (77, 'The Real Thing', 82),
 (78, 'Deixa Entrar', 83),
 (79, 'In Your Honor [Disc 1]', 84),
 (80, 'In Your Honor [Disc 2]', 84),
 (81, 'One By One', 84),
 (82, 'The Colour And The Shape', 84),
 (83, 'My Way: The Best Of Frank Sinatra [Disc 1]', 85),
 (84, 'Roda De Funk', 86),
 (85, 'As Canções de Eu Tu Eles', 27),
 (86, 'Quanta Gente Veio Ver (Live)', 27),
 (87, 'Quanta Gente Veio ver--Bônus De Carnaval', 27),
 (88, 'Faceless', 87),
 (89, 'American Idiot', 54),
 (90, 'Appetite for Destruction', 88),
 (91, 'Use Your Illusion I', 88),
 (92, 'Use Your Illusion II', 88),
 (93, 'Blue Moods', 89),
 (94, 'A Matter of Life and Death', 90),
 (95, 'A Real Dead One', 90),
 (96, 'A Real Live One', 90),
 (97, 'Brave New World', 90),
 (98, 'Dance Of Death', 90),
 (99, 'Fear Of The Dark', 90),
 (100, 'Iron Maiden', 90),
 (101, 'Killers', 90),
 (102, 'Live After Death', 90),
 (103, 'Live At Donington 1992 (Disc 1)', 90),
 (104, 'Live At Donington 1992 (Disc 2)', 90),
 (105, 'No Prayer For The Dying', 90),
 (106, 'Piece Of Mind', 90),
 (107, 'Powerslave', 90),
 (108, 'Rock In Rio [CD1]', 90),
 (109, 'Rock In Rio [CD2]', 90),
 (110, 'Seventh Son of a Seventh Son', 90),
 (111, 'Somewhere in Time', 90),
 (112, 'The Number of The Beast', 90),
 (113, 'The X Factor', 90),
 (114, 'Virtual XI', 90),
 (115, 'Sex Machine', 91),
 (116, 

Question 9
All of the questions in this quiz refer to the open source Chinook Database. Please familiarize yourself with the ER diagram to familiarize yourself with the table and column names to write accurate queries and get the appropriate answers.

Run Query: Find the albums with 12 or more tracks.

In [74]:
%%sql
SELECT * FROM Track LIMIT 100;

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(1, 'For Those About To Rock (We Salute You)', 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 343719, 11170334, 0.99),
 (2, 'Balls to the Wall', 2, 2, 1, None, 342562, 5510424, 0.99),
 (3, 'Fast As a Shark', 3, 2, 1, 'F. Baltes, S. Kaufman, U. Dirkscneider & W. Hoffman', 230619, 3990994, 0.99),
 (4, 'Restless and Wild', 3, 2, 1, 'F. Baltes, R.A. Smith-Diesel, S. Kaufman, U. Dirkscneider & W. Hoffman', 252051, 4331779, 0.99),
 (5, 'Princess of the Dawn', 3, 2, 1, 'Deaffy & R.A. Smith-Diesel', 375418, 6290521, 0.99),
 (6, 'Put The Finger On You', 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 205662, 6713451, 0.99),
 (7, "Let's Get It Up", 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 233926, 7636561, 0.99),
 (8, 'Inject The Venom', 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 210834, 6852860, 0.99),
 (9, 'Snowballed', 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 203102, 6599424, 0.99),
 (10, 'Evil Walks', 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 263497, 8611245, 0.99),
 (11, 'C.O.D.', 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 199836, 6566314, 0.99),
 (12, 'Breaking The Rules', 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 263288, 8596840, 0.99),
 (13, 'Night Of The Long Knives', 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 205688, 6706347, 0.99),
 (14, 'Spellbound', 1, 1, 1, 'Angus Young, Malcolm Young, Brian Johnson', 270863, 8817038, 0.99),
 (15, 'Go Down', 4, 1, 1, 'AC/DC', 331180, 10847611, 0.99),
 (16, 'Dog Eat Dog', 4, 1, 1, 'AC/DC', 215196, 7032162, 0.99),
 (17, 'Let There Be Rock', 4, 1, 1, 'AC/DC', 366654, 12021261, 0.99),
 (18, 'Bad Boy Boogie', 4, 1, 1, 'AC/DC', 267728, 8776140, 0.99),
 (19, 'Problem Child', 4, 1, 1, 'AC/DC', 325041, 10617116, 0.99),
 (20, 'Overdose', 4, 1, 1, 'AC/DC', 369319, 12066294, 0.99),
 (21, "Hell Ain't A Bad Place To Be", 4, 1, 1, 'AC/DC', 254380, 8331286, 0.99),
 (22, 'Whole Lotta Rosie', 4, 1, 1, 'AC/DC', 323761, 10547154, 0.99),
 (23, 'Walk On Water', 5, 1, 1, 'Steven Tyler, Joe Perry, Jack Blades, Tommy Shaw', 295680, 9719579, 0.99),
 (24, 'Love In An Elevator', 5, 1, 1, 'Steven Tyler, Joe Perry', 321828, 10552051, 0.99),
 (25, 'Rag Doll', 5, 1, 1, 'Steven Tyler, Joe Perry, Jim Vallance, Holly Knight', 264698, 8675345, 0.99),
 (26, 'What It Takes', 5, 1, 1, 'Steven Tyler, Joe Perry, Desmond Child', 310622, 10144730, 0.99),
 (27, 'Dude (Looks Like A Lady)', 5, 1, 1, 'Steven Tyler, Joe Perry, Desmond Child', 264855, 8679940, 0.99),
 (28, "Janie's Got A Gun", 5, 1, 1, 'Steven Tyler, Tom Hamilton', 330736, 10869391, 0.99),
 (29, "Cryin'", 5, 1, 1, 'Steven Tyler, Joe Perry, Taylor Rhodes', 309263, 10056995, 0.99),
 (30, 'Amazing', 5, 1, 1, 'Steven Tyler, Richie Supa', 356519, 11616195, 0.99),
 (31, 'Blind Man', 5, 1, 1, 'Steven Tyler, Joe Perry, Taylor Rhodes', 240718, 7877453, 0.99),
 (32, 'Deuces Are Wild', 5, 1, 1, 'Steven Tyler, Jim Vallance', 215875, 7074167, 0.99),
 (33, 'The Other Side', 5, 1, 1, 'Steven Tyler, Jim Vallance', 244375, 7983270, 0.99),
 (34, 'Crazy', 5, 1, 1, 'Steven Tyler, Joe Perry, Desmond Child', 316656, 10402398, 0.99),
 (35, 'Eat The Rich', 5, 1, 1, 'Steven Tyler, Joe Perry, Jim Vallance', 251036, 8262039, 0.99),
 (36, 'Angel', 5, 1, 1, 'Steven Tyler, Desmond Child', 307617, 9989331, 0.99),
 (37, "Livin' On The Edge", 5, 1, 1, 'Steven Tyler, Joe Perry, Mark Hudson', 381231, 12374569, 0.99),
 (38, 'All I Really Want', 6, 1, 1, 'Alanis Morissette & Glenn Ballard', 284891, 9375567, 0.99),
 (39, 'You Oughta Know', 6, 1, 1, 'Alanis Morissette & Glenn Ballard', 249234, 8196916, 0.99),
 (40, 'Perfect', 6, 1, 1, 'Alanis Morissette & Glenn Ballard', 188133, 6145404, 0.99),
 (41, 'Hand In My Pocket', 6, 1, 1, 'Alanis Morissette & Glenn Ballard', 221570, 7224246, 0.99),
 (42, 'Right Through You', 6, 1, 1, 'Alanis Morissette & Glenn Ballard', 176117, 5793082, 0.99),
 (43, 'Forgiven', 6, 1, 1, 'Alanis Morissette & Glenn Ballard', 300355, 9753256, 0.99),
 (44, 'You Learn', 6, 1, 1, 'Alanis Morisset

In [97]:
%%sql
SELECT Album.AlbumId,Album.Title, COUNT(Track.TrackId) as TRACKS FROM Album join Track ON Album.AlbumId = Track.AlbumId GROUP BY Album.AlbumId HAVING TRACKS>=12

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(5, 'Big Ones', 15),
 (6, 'Jagged Little Pill', 13),
 (7, 'Facelift', 12),
 (8, 'Warner 25 Anos', 14),
 (10, 'Audioslave', 14),
 (11, 'Out Of Exile', 12),
 (12, 'BackBeat Soundtrack', 12),
 (14, 'Alcohol Fueled Brewtality Live! [Disc 1]', 13),
 (18, 'Body Count', 17),
 (21, 'Prenda Minha', 18),
 (23, 'Minha Historia', 34),
 (24, 'Afrociberdelia', 23),
 (25, 'Da Lama Ao Caos', 13),
 (26, 'Acústico MTV [Live]', 17),
 (27, 'Cidade Negra - Hits', 14),
 (29, 'Axé Bahia 2001', 14),
 (30, 'BBC Sessions [Disc 1] [Live]', 14),
 (32, 'Carnaval 2001', 14),
 (33, 'Chill: Brazil (Disc 1)', 17),
 (34, 'Chill: Brazil (Disc 2)', 17),
 (36, 'Greatest Hits II', 17),
 (37, 'Greatest Kiss', 20),
 (38, 'Heart of the Night', 12),
 (39, 'International Superhits', 21),
 (40, 'Into The Light', 12),
 (41, 'Meus Momentos', 14),
 (42, 'Minha História', 14),
 (45, 'Sambas De Enredo 2001', 14),
 (46, 'Supernatural', 13),
 (47, 'The Best of Ed Motta', 14),
 (48, 'The Essential Miles Davis [Disc 1]', 13),
 (51, "Up An' Atom", 22),
 (52, 'Vinícius De Moraes - Sem Limite', 15),
 (53, 'Vozes do MPB', 14),
 (54, 'Chronicle, Vol. 1', 20),
 (55, 'Chronicle, Vol. 2', 20),
 (56, 'Cássia Eller - Coleção Sem Limite [Disc 2]', 15),
 (57, 'Cássia Eller - Sem Limite [Disc 1]', 15),
 (63, 'Purpendicular', 12),
 (67, "Vault: Def Leppard's Greatest Hits", 16),
 (69, 'Djavan Ao Vivo - Vol. 02', 13),
 (70, 'Djavan Ao Vivo - Vol. 1', 13),
 (71, 'Elis Regina-Minha História', 14),
 (72, 'The Cream Of Clapton', 18),
 (73, 'Unplugged', 30),
 (74, 'Album Of The Year', 12),
 (75, 'Angel Dust', 14),
 (76, 'King For A Day Fool For A Lifetime', 15),
 (78, 'Deixa Entrar', 14),
 (82, 'The Colour And The Shape', 13),
 (83, 'My Way: The Best Of Frank Sinatra [Disc 1]', 24),
 (84, 'Roda De Funk', 16),
 (85, 'As Canções de Eu Tu Eles', 14),
 (86, 'Quanta Gente Veio Ver (Live)', 15),
 (88, 'Faceless', 12),
 (89, 'American Idiot', 13),
 (90, 'Appetite for Destruction', 12),
 (91, 'Use Your Illusion I', 16),
 (92, 'Use Your Illusion II', 14),
 (93, 'Blue Moods', 13),
 (95, 'A Real Dead One', 12),
 (99, 'Fear Of The Dark', 12),
 (102, 'Live After Death', 18),
 (115, 'Sex Machine', 20),
 (119, 'Get Born', 13),
 (120, 'Are You Experienced?', 17),
 (122, 'Jorge Ben Jor 25 Anos', 14),
 (123, 'Jota Quest-1995', 12),
 (124, 'Cafezinho', 14),
 (125, 'Living After Midnight', 16),
 (126, 'Unplugged [Live]', 15),
 (139, 'A TempestadeTempestade Ou O Livro Dos Dias', 15),
 (140, 'Mais Do Mesmo', 16),
 (141, 'Greatest Hits', 57),
 (142, 'Lulu Santos - RCA 100 Anos De Música - Álbum 01', 14),
 (143, 'Lulu Santos - RCA 100 Anos De Música - Álbum 02', 14),
 (145, 'Barulhinho Bom', 18),
 (146, 'Seek And Shall Find: More Of The Best (1963-1981)', 18),
 (148, 'Black Album', 12),
 (149, 'Garage Inc. (Disc 2)', 16),
 (151, 'Load', 14),
 (153, 'ReLoad', 13),
 (157, 'Miles Ahead', 14),
 (158, 'Milton Nascimento Ao Vivo', 13),
 (159, 'Minas', 13),
 (160, 'Ace Of Spades', 15),
 (161, 'Demorou...', 12),
 (162, 'Motley Crue Greatest Hits', 17),
 (163, 'From The Muddy Banks Of The Wishkah [Live]', 17),
 (164, 'Nevermind', 12),
 (165, 'Compositores', 15),
 (166, 'Olodum', 14),
 (167, 'Acústico MTV', 21),
 (168, 'Arquivo II', 12),
 (169, 'Arquivo Os Paralamas Do Sucesso', 16),
 (174, 'Tribute', 14),
 (175, 'Walking Into Clarksdale', 12),
 (176, 'Original Soundtracks 1', 14),
 (178, 'Live On Two Legs [Live]', 16),
 (179, 'Pearl Jam', 13),
 (180, 'Riot Act', 15),
 (182, 'Vs.', 12),
 (184, 'Os Cães Ladram Mas A Caravana Não Pára', 16),
 (185, 'Greatest Hits I', 17),
 (189, 'New Adventures In Hi-Fi', 14),
 (190, 'The Best Of R.E.M.: The IRS Years', 16),
 (192, 'Raul Seixas', 14),
 (193, 'Blood Sugar Sex Magik', 17),
 (194, 'By The Way', 16),
 (195, 'Californication', 15),
 (196, 'Retrospective I (1974-1980)', 14),
 (199, 'Maquinarama', 12),
 (201, 'Judas 0: B-Sides and Rarities', 16),
 (202, 'Rotten Apples: Greatest Hits', 18),
 (203, 'A-Sides', 17),
 (206, 'Core', 12),
 (211, 'The Singles', 18),
 (212, 'Beyond Good And Evil', 

one table

In [105]:
%%sql

SELECT AlbumId, COUNT(AlbumId) AS Tracks FROM Track Group BY AlbumId HAVING COUNT(AlbumId)>=12

 * sqlite:///Chinook_Sqlite.sqlite
Done.


[(5, 15),
 (6, 13),
 (7, 12),
 (8, 14),
 (10, 14),
 (11, 12),
 (12, 12),
 (14, 13),
 (18, 17),
 (21, 18),
 (23, 34),
 (24, 23),
 (25, 13),
 (26, 17),
 (27, 14),
 (29, 14),
 (30, 14),
 (32, 14),
 (33, 17),
 (34, 17),
 (36, 17),
 (37, 20),
 (38, 12),
 (39, 21),
 (40, 12),
 (41, 14),
 (42, 14),
 (45, 14),
 (46, 13),
 (47, 14),
 (48, 13),
 (51, 22),
 (52, 15),
 (53, 14),
 (54, 20),
 (55, 20),
 (56, 15),
 (57, 15),
 (63, 12),
 (67, 16),
 (69, 13),
 (70, 13),
 (71, 14),
 (72, 18),
 (73, 30),
 (74, 12),
 (75, 14),
 (76, 15),
 (78, 14),
 (82, 13),
 (83, 24),
 (84, 16),
 (85, 14),
 (86, 15),
 (88, 12),
 (89, 13),
 (90, 12),
 (91, 16),
 (92, 14),
 (93, 13),
 (95, 12),
 (99, 12),
 (102, 18),
 (115, 20),
 (119, 13),
 (120, 17),
 (122, 14),
 (123, 12),
 (124, 14),
 (125, 16),
 (126, 15),
 (139, 15),
 (140, 16),
 (141, 57),
 (142, 14),
 (143, 14),
 (145, 18),
 (146, 18),
 (148, 12),
 (149, 16),
 (151, 14),
 (153, 13),
 (157, 14),
 (158, 13),
 (159, 13),
 (160, 15),
 (161, 12),
 (162, 17),
 (163, 17),
 (164, 12),
 (165, 15),
 (166, 14),
 (167, 21),
 (168, 12),
 (169, 16),
 (174, 14),
 (175, 12),
 (176, 14),
 (178, 16),
 (179, 13),
 (180, 15),
 (182, 12),
 (184, 16),
 (185, 17),
 (189, 14),
 (190, 16),
 (192, 14),
 (193, 17),
 (194, 16),
 (195, 15),
 (196, 14),
 (199, 12),
 (201, 16),
 (202, 18),
 (203, 17),
 (206, 12),
 (211, 18),
 (212, 12),
 (213, 18),
 (215, 14),
 (216, 12),
 (217, 14),
 (218, 15),
 (219, 15),
 (221, 20),
 (222, 15),
 (223, 15),
 (224, 22),
 (225, 16),
 (227, 19),
 (228, 23),
 (229, 26),
 (230, 25),
 (231, 24),
 (232, 12),
 (234, 15),
 (236, 12),
 (237, 17),
 (238, 14),
 (241, 14),
 (242, 12),
 (243, 17),
 (245, 12),
 (246, 13),
 (247, 15),
 (248, 19),
 (250, 22),
 (251, 25),
 (253, 24),
 (255, 23),
 (256, 12),
 (257, 12),
 (258, 19),
 (259, 17),
 (261, 17),
 (270, 14),
 (271, 14),
 (321, 12)]